### PR TITLE
Fix confusion over what function is called when delivering mailing

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix confusion over what function is called when delivering mailing

Before
----------------------------------------
What happens in `MailingJob::runJobs` is impossible to determine from the code - however historical knowledge tells us that it calls flexmaler unless it is being used for SMS - in which case it calls the `deliver()` function  -which is ONLY used for SMS

After
----------------------------------------
The IF on the `deliver()` function is replace with a mode check

Now that it is clear the deliver function is only called by the child SMS class we can move it from the parent to the child & remove the also-never-used parent copy of `deliverGroup()` which was already being overriden by the copy on the SMS child class in all actual usages

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by pradpnayak** [comment]
This looks good to me

**Comment by eileenmcnaughton** [comment]
Thanks @pradpnayak for checking it out!

File: CRM/Mailing/BAO/MailingJob.php [hunk 1]
```diff
@@ -157,20 +157,13 @@ public static function runJobs($testParams = NULL, $mode = NULL) {
         $transaction->commit();
       }
 
-      // Get the mailer
+      // Compose and deliver each child job
       if ($mode === NULL) {
         $mailer = \Civi::service('pear_mail');
```
Comment (seamuslee001) [inline CRM/Mailing/BAO/MailingJob.php:162]: @eileenmcnaughton can this go as well?

Comment (seamuslee001) [reply CRM/Mailing/BAO/MailingJob.php:162]: actually sorry just saw where it is being used don't worry

**Pull request merged**